src/map.h

```diff
@@ -257,9 +259,43 @@ class Map /*: public NodeContainer*/
 	void removeNodeTimer(v3s16 p);
 
 	/*
-		Variables
+		Utilities
 	*/
 
+	// Iterates through all nodes in the area in an unspecified order.
+	// The given callback takes the position as its first argument and the node
+	// as its second. If it returns false, forEachNodeInArea returns early.
+	template<typename F>
+	void forEachNodeInArea(v3s16 minp, v3s16 maxp, F func)
+	{
+		v3s16 bpmin = getNodeBlockPos(minp);
+		v3s16 bpmax = getNodeBlockPos(maxp);
+		for (s16 bz = bpmin.Z; bz <= bpmax.Z; bz++)
+		for (s16 bx = bpmin.X; bx <= bpmax.X; bx++)
+		for (s16 by = bpmin.Y; by <= bpmax.Y; by++) {
+			// y is iterated innermost to make use of the sector cache.
+			v3s16 bp(bx, by, bz);
+			MapBlock *block = getBlockNoCreateNoEx(bp);
```

Comment:
can't this return null?

Reply:
That's checked in the loop, though I'm sure the check can be lifted out of the loop: https://github.com/TurkeyMcMac/minetest/blob/c75b084b9569bf2bb83ba8f2eee85d08814f6623/src/map.h#L290-L292

Reply:
That is, the compiler can lift the check out of the loop.